### DISCUSSION TITLE
How to migrate MLFlow from v1 to v2?

### DISCUSSION THREAD

**Discussion by tamis-laan** [mlflow/mlflow#8206] [Q&A]
So we have MLFlow version 1.3 running and would like to migrate to version 2. **Do we need to do a migration?**

**Comment by JHibbard** [comment]
Hi tamis-laan,

Great question! The tracking server's database has had schema changes to accommodate new features over time. To update your tracking database you'll have to perform a database upgrade, documentation [here](https://github.com/mlflow/mlflow/tree/master/mlflow/store/db_migrations). There may be other migration consideration, e.g. if you're proxying your artifact storage the links will be broken if your new setup isn't proxied in the same way:

> When an experiment is created, the artifact storage location from the configuration of the tracking server is logged in the experiment’s metadata. When enabling proxied artifact storage, any existing experiments that were created while operating a tracking server in non-proxied mode will continue to use a non-proxied artifact location. In order to use proxied artifact logging, a new experiment must be created. If the intention of enabling a tracking server in -serve-artifacts mode is to eliminate the need for a client to have authentication to the underlying storage, new experiments should be created for use by clients so that the tracking server can handle authentication after this migration.

So to answer fully, it'd be great to get more details on your setup. The database migration will be the main thing I expect, but still back everything up and test :)

Hope that helps!
Jim

**Comment by namiyousef** [comment]
Hi @JHibbard had a question regarding the migration of artefacts? 
I have been able to upgrade the database schema from v1.x.x to v2.x.x, however I'm no longer able to use artefacts, because the version of `cloudpickle` has changed. On my previous version of mlflow (1.21.0), there were no constraints on `cloudpickle` (see: https://github.com/mlflow/mlflow/blob/v1.21.0/setup.py#L45) to I had cloudpickle 3.0.0 installed. From mlflow 1.28.0 onwards, cloudpickle is constrained to <3.0.0. Is there a guide for migration of artefacts due to the change of cloudpickle version ?
